pkg/octaviaapi/deployment.go

```diff
@@ -137,25 +142,25 @@ func Deployment(
 	deployment.Spec.Template.Spec.Affinity = affinity.DistributePods(
 		common.AppSelector,
 		[]string{
-			ServiceName,
+			serviceName,
 		},
 		corev1.LabelHostname,
 	)
 	if instance.Spec.NodeSelector != nil && len(instance.Spec.NodeSelector) > 0 {
 		deployment.Spec.Template.Spec.NodeSelector = instance.Spec.NodeSelector
 	}
 
-	initContainerDetails := APIDetails{
+	initContainerDetails := octavia.APIDetails{
 		ContainerImage:       instance.Spec.ContainerImage,
-		DatabaseHost:         instance.Status.DatabaseHostname,
+		DatabaseHost:         instance.Spec.DatabaseHostname,
```

Comment:
```suggestion
		DatabaseHost:         instance.Status.DatabaseHostname,
```

The change in this line seems to have caused the octavia-api pod to get stuck in Init:CrashLoopBackOff with the error

```
+ export PASSWORD=12345678
+ PASSWORD=12345678
/usr/local/bin/container-scripts/init.sh: line 23: DatabaseHost: Please specify a DatabaseHost variable.
```

Reply:
I'll investigate this. AFAICT this bit of code is not different than what I've been using and has been working fine.

Reply:
I think it's likely that an outdated install_yamls was used. The database hostname is passed to OctaviaAPI by the Octavia instance through it's spec, not it's status.